controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTrigger.java

```diff
@@ -266,19 +269,18 @@ private boolean isCapacityConstrained(JobType jobType) {
     }
 
     /** Returns the first job that has been running for more than the given timeout */
-    private Optional<JobStatus> firstDeadJob(DeploymentJobs jobs, Duration timeout) {
-        Instant startOfGracePeriod = controller.clock().instant().minus(timeout);
+    private Optional<JobStatus> firstDeadJob(DeploymentJobs jobs) {
         Optional<JobStatus> oldestRunningJob = jobs.jobStatus().values().stream()
-                .filter(JobStatus::inProgress)
+                .filter(job -> job.isRunning(Instant.ofEpochMilli(0)))
                 .sorted(Comparator.comparing(status -> status.lastTriggered().get().at()))
                 .findFirst();
-        return oldestRunningJob.filter(job -> job.lastTriggered().get().at().isBefore(startOfGracePeriod));
+        return oldestRunningJob.filter(job -> job.lastTriggered().get().at().isBefore(jobTimeoutLimit()));
     }
 
     /** Decide whether the job should be triggered by the periodic trigger */
     private boolean shouldRetryNow(JobStatus job) {
         if (job.isSuccess()) return false;
-        if (job.inProgress()) return false;
+        if (job.isRunning(jobTimeoutLimit())) return false;
```

Comment:
I think this will cause dead jobs to be retried multiple times, or by the wrong mechanism, same applies to L201. I suggest having an overloaded `isRunning()` where the default `jobTimeoutLimit` is `Instant.MIN` - effectively the same behaviour as `inProgress`.

Reply:
But we don't actually consider a job running if it has been dead for more than 12 hours ...

I think you are right that several of these maintenance jobs can now come to resurrect dead jobs, but that seems harmless to me - just a stepping stone towards replacing them by one.

Reply:
Likely harmless in practice, yes, but it becomes even harder to reason about because the logged trigger cause may be wrong.

Reply:
Yes, but simplification is coming ...